src/build.sh

```diff
@@ -124,6 +128,41 @@ function build_kernel_main()
     return "$?"
   fi
 
+  if [[ -n "$from_sha_arg" ]]; then
+    # Check if there is a rebase in process.
+    if [[ -d .git/rebase-merge ]]; then
+      printf '%s\n' 'ERROR: Abort the repository rebase before continuing with build from sha (use "git rebase --abort")!'
+      return 22 #EINVAL
+    elif [[ -f .git/MERGE_HEAD ]]; then
+      printf '%s\n' 'ERROR: Abort the repository merge before continuing with build from sha (use "git rebase --abort")!'
+      return 22 #EINVAL
+    elif [[ -f .git/BISECT_LOG ]]; then
+      printf '%s\n' 'ERROR: Stop the repository bisect before continuing with build from sha (use "git bisect reset")!'
+      return 22 #EINVAL
+    elif [[ -d .git/rebase-apply ]]; then
+      printf '%s\n' 'ERROR: Abort the repository patch apply before continuing with build from sha (use "git am --abort")!'
+      return 22 #EINVAL
```

Comment:
IMHO, for all the checks above, we get to an error not because the argument is invalid (the meaning of 22 EINVAL), but because we shouldn't try to start a new `git rebase` in any of the cases above. That being said, I don't know an errno that would fit perfectly, but I think that 125 ECANCELED works better.

@rodrigosiqueira, what are your thoughts on this?

Reply:
I agree.

Actually, I'm also thinking about this code. How about creating a new file under `src/lib/` named `git.sh`? We should add checks like this in a library since many parts of kw rely on git, and we constantly reinvent the wheel in different files. @MarceloSpessoto, you don't need to do it here, but it would be nice if you opene an issue to address this and create a PR moving this code to this new lib file.